pontoon/base/views.py

```diff
@@ -377,6 +377,27 @@ def get_translations_from_other_locales(request):
     return JsonResponse(payload, safe=False)
 
 
+@utils.require_AJAX
+def get_sibling_entities(request):
+    """Get entities preceding and succeeding the current entity"""
+    print("TEST-------------------")
+    try:
+        entity = int(request.GET["entity"])
+        locale = request.GET["locale"]
+        print(entity, locale)
+    except (MultiValueDictKeyError, ValueError) as e:
+        return JsonResponse(
+            {"status": False, "message": "Bad Request: {error}".format(error=e)},
+            status=400,
+        )
+
+    entity = get_object_or_404(Entity, pk=entity)
+    locale = get_object_or_404(Locale, code=locale)
+
+    payload = []
+    return JsonResponse(payload, safe=False)
```

Comment:
Great job so far! Now comes the fun part, where we'll actually return the data that we need. :)

Instead of an empty list, we should return a list of preceding entities and a list of succeeding entities. You should use a [`dictionary`](https://docs.python.org/3/tutorial/datastructures.html#dictionaries) with two keys ("preceding" and "succeeding") to combine the lists into one data structure. You'll see we use that pattern a lot in other `JsonResponse` instances.

To generate the actual content for the two lists, you should use `Entity.map_entities()` for each, which returns all the data required to render entities in the string list for the given locale and entities, which is exactly what we need here. See how it's used in `_get_entities_list()` for example.

`Entity.map_entities()` takes 3 arguments:
- `locale`: we already have it
- `preferred_source_locale`: see the `entities()` view to see how to define it
- `entities`: we'll need an `Entity` [QuerySet](https://docs.djangoproject.com/en/3.1/ref/models/querysets/) that gets the relevant data from the DB. See below.

1. You'll need to [filter](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#filter) entities for which `resource` (i.e. file) equals the resource of the entity the user has clicked on, and for which the `obsolete` flag is set to `False` (i.e. the entity hasn't been removed).
2. Then you'll need to [`order_by`](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#django.db.models.query.QuerySet.order_by) the `order` field and again filter entities for which `order` value is less (for preceding) or more (for succeeding) than the order of the given entity.
3. Finally, you'll only need to retrieve the first two matching instances, which you can do by simply adding `[:2]` at the end of the QuerySet.

Note that filter() and order_by() expressions can be chained, so you can do `Entity.objects.filter(...).order_by(...).filter(...)...`.